collect_app/src/main/java/org/odk/collect/android/preferences/ServerPreferencesFragment.java

```diff
@@ -490,4 +490,21 @@ private void continueOnBackPressed() {
     public void doBack() {
         runGoogleAccountValidation();
     }
+
+    @Override
+    public void onResume() {
+        super.onResume();
+        serverUrlPreference.setOnPreferenceClickListener(preference -> {
```

Comment:
This code looks right to me but I think these listeners should be setup in `addAggregatePreferences` along with the rest of the setup for each `Preference`.

Reply:
@grzesiek2010, Yes I had checked that  before pull request and that is the default behaviour

![img](https://user-images.githubusercontent.com/44516722/72404404-38b34800-377b-11ea-8554-c21d1b18faf9.png)

So, Should I remove the code?

Reply:
Yuck that's annoying. I'm not seeing that problem in Android settings but maybe they work around it as well. 

Personally I think it makes sense to keep these changes as it's definitely a better experience (in the most basic sense it cuts down on taps). What do you think @grzesiek2010?

Reply:
I'm not sure but also not strongly against so you can proceed @Vivever but as I said we have more such `EditTextPreferences` to fix.

Reply:
@grzesiek2010, I will find them and fix them.

Reply:
@grzesiek2010 @seadowg  There was only one extra EditTextPreference (Admin Password in Admin Settings) which I have fixed. 
Other preferences are either checkbox or radio buttons.

Reply:
Thanks for looking at that @Vivever! As far as I can see there are still a few fields in "User and device identity" > "Form metadata".

As I said before it'd be better not to call `setPreferenceClickListener` in `onResume`. These should probably go with the rest of the preference setup in `ServerPreferencesFragment#addAggregatePreferences`. In other fragments it can just be in `onCreate`. You shouldn't need to `setPreferenceClickListener(null)` either as the preference object won't (as far as I know) outlive the Fragment in memory.